app/controllers/giveaways_controller.rb

```diff
@@ -1,21 +1,22 @@
 # frozen_string_literal: true
 
 class GiveawaysController < ApplicationController
+  before_action :must_be_logged_in, only: %i[new edit destroy]
```

Comment:
I believe we also need to make sure the user is logged in for `create` and `update` actions.

Reply:
@violeta-p That's true, I was just targeting the GET actions, for which one can access the path through typing the URL in the bar. I don't know if it is possible to make an POST or PATCH via URL string?